PullToRefresh/UIScrollView+PullToRefresh.swift

```diff
@@ -10,43 +10,81 @@ import Foundation
+private var bottomPullToRefreshKey: UInt8 = 0
 
 public extension UIScrollView {
-    private(set) var pullToRefresh: PullToRefresh? {
+    private(set) var topPullToRefresh: PullToRefresh? {
         get {
-            return objc_getAssociatedObject(self, &associatedObjectHandle) as? PullToRefresh
+            return objc_getAssociatedObject(self, &topPullToRefreshKey) as? PullToRefresh
         }
         set {
-            objc_setAssociatedObject(self, &associatedObjectHandle, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+            objc_setAssociatedObject(self, &topPullToRefreshKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
         }
     }
     
-    public func addPullToRefresh(pullToRefresh: PullToRefresh, action:()->()) {
-        if self.pullToRefresh != nil {
-            self.removePullToRefresh(self.pullToRefresh!)
+    private(set) var bottomPullToRefresh: PullToRefresh? {
+        get {
+            return objc_getAssociatedObject(self, &bottomPullToRefreshKey) as? PullToRefresh
         }
-        
-        self.pullToRefresh = pullToRefresh
+        set {
+            objc_setAssociatedObject(self, &bottomPullToRefreshKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+    
+    public func addPullToRefresh(pullToRefresh: PullToRefresh, action:()->()) {
         pullToRefresh.scrollView = self
         pullToRefresh.action = action
         
+        var originY: CGFloat
         let view = pullToRefresh.refreshView
-        view.frame = CGRectMake(0, -view.frame.size.height, self.frame.size.width, view.frame.size.height)
+        switch pullToRefresh.position {
+        case .Top:
+            if self.topPullToRefresh != nil {
+                self.removePullToRefresh(self.topPullToRefresh!)
+            }
+            self.topPullToRefresh = pullToRefresh
+            originY = -view.frame.size.height
+        case .Bottom:
+            if self.bottomPullToRefresh != nil {
+                self.removePullToRefresh(self.bottomPullToRefresh!)
+            }
+            self.bottomPullToRefresh = pullToRefresh
+            originY = self.contentSize.height
+        }
+        
+        view.frame = CGRectMake(0, originY, self.frame.size.width, view.frame.size.height)
         self.addSubview(view)
         self.sendSubviewToBack(view)
     }
     
     func removePullToRefresh(pullToRefresh: PullToRefresh) {
-        self.pullToRefresh?.refreshView.removeFromSuperview()
-        self.pullToRefresh = nil
+        switch pullToRefresh.position {
+        case .Top:
+            self.topPullToRefresh?.refreshView.removeFromSuperview()
+            self.topPullToRefresh = nil
+        case .Bottom:
+            self.bottomPullToRefresh?.refreshView.removeFromSuperview()
+            self.bottomPullToRefresh = nil
+        }
     }
     
-    func startRefreshing() {
-        pullToRefresh?.startRefreshing()
+    func startRefreshing(position pos: Position = .Top) {
```

Comment:
We don't use short names, please replace with: `func startRefreshing(at position: Position = .Top)`